src/js/libgeo/Prover.js

```diff
@@ -68,21 +74,111 @@ function incidentPC(p, c) {
             var erg = General.mult(c.matrix, p.homog);
             erg = General.mult(p.homog, erg);
             erg = CSNumber.abs(erg);
-            if (erg.value.real < 0.0000000000001) {
-                p.incidences.push(c.name);
-                c.incidences.push(p.name);
-            }
+            return (erg.value.real < 0.0000000000001);
         }
     };
 }
 
 function checkConjectures() {
+    var debug = false;
+    if (debug) console.log("conjectures", conjectures.length);
     if (conjectures.length === 0) return;
-    // TODO: we need some randomized proving here:
-    // move free elements and check conjectures a number of times.
-    // For now assume that all conjectures could be verified.
+    //backupGeo
+    stateArrays.prover.set(stateIn);
+
+    var nummoves = 3;
+
+    // filter free objects which are involved in conjectures
+    var involved;
+
+    var recalcInvolved = function() {
+        involved = [];
+        conjectures.forEach(function(con) {
+            var invs = con.getInvolved();
+            var incis;
+            invs.forEach(function(el) {
+                if (involved.indexOf(el) < 0) {
+                    involved.push(el);
+                    // also add incidences of involved objects
+                    incis = findAllIncis(el, []);
+                    incis.forEach(function(i) {
+                        if (involved.indexOf(i) < 0) involved.push(i);
+                    });
+                }
+            });
+        });
+    };
+
+    // recursively find all incidences to an geo object
+    var findAllIncis = function(el, list) {
+        // get all incidences we don't already know
+        var nincis = el.incidences.filter(function(iels) {
+            return list.indexOf(csgeo.csnames[iels]) < 0;
+        });
+        if (nincis.length === 0) return list;
+        // add new incidences
+        nincis.forEach(function(ii) {
+            list.push(csgeo.csnames[ii]);
+        });
+        // recursive call
+        nincis.forEach(function(nel) {
+            return findAllIncis(csgeo.csnames[nel], list);
+        });
+        return list;
```

Comment:
same here, `list` as dict would probably faster and readable (but change the name before)